java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -72,4 +83,168 @@ public void ping_with_message(BaseClient client) {
         String data = client.ping("H3LL0").get();
         assertEquals("H3LL0", data);
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void set_and_get_without_options(BaseClient client) {
+        String ok = client.set(KEY_NAME, INITIAL_VALUE).get();
+        assertEquals(OK, ok);
+
+        String data = client.get(KEY_NAME).get();
+        assertEquals(INITIAL_VALUE, data);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void get_missing_value(BaseClient client) {
+        String data = client.get("invalid").get();
+        assertNull(data);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void set_overwrite_value_and_returnOldValue_returns_string(BaseClient client) {
+        String ok = client.set(KEY_NAME, INITIAL_VALUE).get();
+        assertEquals(OK, ok);
+
+        SetOptions options = SetOptions.builder().returnOldValue(true).build();
+        String data = client.set(KEY_NAME, ANOTHER_VALUE, options).get();
+        assertEquals(INITIAL_VALUE, data);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void set_requires_a_value(BaseClient client) {
+        assertThrows(NullPointerException.class, () -> client.set("SET", null));
```

Comment:
minor - the null exception tests can be in UTs, not ITs.

Reply:
San can move these to the next PR?